src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ArrayAdapter implements CacheItemPoolInterface
```

Comment:
@nicolas-grekas Before considering to open pr on documenting this adapter better, what is your's intention behind this one?

A. Do you see it only for testing, hence we should clarify and avoid using it in doc examples. 
B. Or do you think it should be supported in prod using a chain proxy in front of an actual cache pool for performance? Meaning it will need to handle limits to avoid using all memory, and will also need to reduce risk of returning stale data while also documenting that there will always be a risk of returning stale data *(unless we add flag to getItems to on a case by case basis let users specify if they accept stale data or not, default being not)*.

Context:
Stash has issues with this, the adapter is php documented as for unit tests only, but the bundle exposes an option to enable "InMemory" cache, w/o further details on the risk of broken application logic it might cause.
See alos: https://github.com/php-cache/array-adapter/issues/3